packages/test/test-end-to-end-tests/src/test/gc/gcSweepDataStores.spec.ts

```diff
@@ -630,56 +642,65 @@ describeCompat("GC data store sweep tests", "NoCompat", (getTestObjectProvider)
 			},
 		);
 
-		it("trailing ops cause sweep ready data store to be realized by summarizer", async () => {
-			const container = await provider.makeTestContainer(testContainerConfig);
-			const defaultDataObject = (await container.getEntryPoint()) as ITestDataObject;
-			await waitForContainerConnection(container);
-
-			// Create a data store and make it unreferenced.
-			const ds2key = "ds2";
-			const dataStore =
-				await defaultDataObject._context.containerRuntime.createDataStore(TestDataObjectType);
-			const testDataObject =
-				await getDataStoreEntryPointBackCompat<ITestDataObject>(dataStore);
-			assert(
-				testDataObject !== undefined,
-				"Should have been able to retrieve testDataObject from entryPoint",
-			);
-			const unreferencedId = testDataObject._context.id;
-			defaultDataObject._root.set(ds2key, testDataObject.handle);
-			defaultDataObject._root.delete(ds2key);
-
-			// Summarize so that the above data store is unreferenced.
-			const { summarizer: summarizer1 } = await loadSummarizer(container);
-			const { summaryVersion } = await ensureSynchronizedAndSummarize(summarizer1);
-
-			// Send a trailing op and close the summarizer.
-			testDataObject._root.set("key", "value");
-			summarizer1.close();
-			await delay(sweepTimeoutMs + 10);
-
-			const logger = new MockLogger();
-			// Close the container because in real scenarios, session expiry will close it. Not closing it
-			// will cause GC_Deleted_DataStore_Unexpected_Delete error.
-			container.close();
-			// Summarize. The tombstone ready data store should get realized because it has a
-			// trailing op.
-			const { summarizer: summarizer2 } = await loadSummarizer(
-				container,
-				summaryVersion,
-				logger,
-			);
-			await assert.doesNotReject(
-				ensureSynchronizedAndSummarize(summarizer2),
-				"summarize failed",
-			);
-			logger.assertMatch([
+		itExpects(
+			"trailing ops cause sweep ready data store to be realized by summarizer",
+			[
 				{
-					eventName: "fluid:telemetry:Summarizer:Running:SweepReadyObject_Realized",
-					id: { value: `/${unreferencedId}`, tag: TelemetryDataTag.CodeArtifact },
+					eventName:
```

Comment:
The workaround to avoid this was to close the container that created the data store being deleted. However, on removing the delays, the test failied and I realized that closing the container causes the trailing op it sent to be not picked up by the `LoaderContainerTracker`. So, when the next summary happens, `ensureSynchronized` does not wait for the trailing op.

This should not have worked before this change either but for some reason it does maybe due to the awaits. My theory is that the op is likely processed by the summarizer by the time ensureSynchronized is called.

Reply:
So, in this change I removed closing the container and added this itExpects. I think we can remove logging the `GC_Deleted_DataStore_Unexpected_Delete` error.

Reply:
Could you have added a flush to the outbound queue or something?  Like `provider.processOutgoing`?  That waits for all ops to be sequenced and received (but not processed) by all tracked containers.

Also, I don't think we should remove that log.  It's an important assumption that we should protect, so it doesn't accidentally regress with any of the architectural changes that may come to the runtime in the next year+.

Reply:
While incorporating https://github.com/microsoft/FluidFramework/pull/23516#discussion_r1913760120, I used a workaround to not do this.